frappe/boot.py

```diff
@@ -95,20 +95,25 @@ def get_allowed_pages():
 
 def get_allowed_reports():
 	return get_user_page_or_report('Report')
-	
+
 def get_user_page_or_report(parent):
 	roles = frappe.get_roles()
 	has_role = {}
 	column = get_column(parent)
-	
+	report_type_col = ""
+	if parent == "Report":
+		report_type_col = ", `tabCustom Role`.report_type"
+
 	# get pages or reports set on custom role
-	for p in frappe.db.sql("""select `tabCustom Role`.{field} as name, `tabCustom Role`.modified, `tabCustom Role`.ref_doctype
+	for p in frappe.db.sql("""select `tabCustom Role`.{field} as name, `tabCustom Role`.modified, `tabCustom Role`.ref_doctype {report_type_col}
 		from `tabCustom Role`, `tabHas Role` where
 			`tabHas Role`.parent = `tabCustom Role`.name and
 			`tabCustom Role`.{field} is not null and `tabHas Role`.role in ({roles})
 			""".format(field=parent.lower(), roles = ', '.join(['%s']*len(roles))), roles, as_dict=1):
 
 		has_role[p.name] = {"modified":p.modified, "title": p.name, "ref_doctype": p.ref_doctype}
+		if parent == "Report":
+			has_role[p.name].update({'report_type': p.report_type})
```

Comment:
Used to set routes for awesome bar options, differentiating by "Report Builder" or other type.

![screen shot 2017-04-13 at 2 17 17 pm](https://cloud.githubusercontent.com/assets/5196925/24997339/eff243dc-2053-11e7-9c58-71e42cbfdfe1.png)

Was default behaviour. Problem came up on support issue.